src/Schema/Keywords/MultipleOf.php

```diff
@@ -25,9 +24,9 @@ class MultipleOf extends BaseKeyword
     public function validate($data, $multipleOf): void
     {
         try {
-            Validator::numeric()->assert($data);
-            Validator::numeric()->positive()->assert($multipleOf);
-        } catch (ExceptionInterface $e) {
+            Validator::numericVal()->assert($data);
```

Comment:
I don't see real changes other than `ExceptionInterace` -> `Exception` and `numeric` -> `numericVal`. I think this can be tried to overcome with some kind of compatibility hacks to support both versions

Reply:
We shouldn't even need a hack this should work:
```
} catch (ExceptionInterface|\Respect\Validation\Exceptions\Exception $e) {
```

php doesn't care if the class your matching exists or not

Reply:
@franksl can you please try to update catch blocks instead of replace it? Also with imports instead of FQCN use? I  will think about `numeric` meanwhile

Reply:
actually one way to solve it is to ask `numericVal` into 1.x version as an alias to `numeric`

Reply:
> php doesn't care if the class your matching exists or not

I strongly disagree, sorry but if php lets you write names of classes that don't exist to me is a BIG flaw. I absolutely don't mean to be rude but taking advantage of what I consider flaws is not the way I would fix that.

Reply:
Do you have any other way to support both version then?

My fix is simple and it's supported by php so I don't see a problem with that.

Reply:
If all the methods of respect/validation used by this library work on php 7.2 I would support only validation v2.
It's even simpler. 
Personal opinion: the fact that php supports what I consider bad practices does not mean that they should be encouraged. Surely I will not put my name on that. The easiest way is not always the right way.

Reply:
But that's not the problem. 

Lets say I'm using respect/validation v1 and this library. 
Your way of doing this breaks this for me because I can't upgrade because of blocking requirement on v2. 

If we support both 1 and 2 it would install v1 for me and v2 for you. 

And the library would continue to work for php7.2

Reply:
> I would support only validation v2. It's even simpler.

This means than next day after rease we got a bug report for older release and we say "oh, dude, you need to update your php, half of your project and a lot more in order to get this bugfix because now we support only respect v2".

Other way is to support two versions simulataniously, which is a lot overhead, since then we should do this for each library which get a major update.

So the easiest way is to make any assumptions of your dependencies and make different code work for different parts of them.

Imaging having these code in C
```c
#ifdef REPSECT_VERSION_2
  do_some_new_stuff(args);
#else
 do_some_old_stuff(args);
#endif
```

Having class loader lazy is the only way to have optional dependencies. Otherwize any framework will carry all the dependencies it can support or we will get a hook hell for configuring plugin system (imagine any service would be a drupal with hundred of plugins)

Reply:
Correct, but considering that the last release of respect/validation is 1 year and a half old I must assume that the v1 branch is highly abandoned in favor of the new one.
If the purpose of this library is to support all the old versions of all its dependencies well... good luck... this is not my home, I don't make any decisions, I'm just suggesting an upgrade that, to me, is reasonable.

Reply:
If we are talking about end-user project (like complete system with business logic and so on) I'm totally agree with you. Doing weird stuff with classloader is ugly and is not welcome. But main purpose of the library is to provider smooth DX to the programmers who use it. And the never know whether we use classloader trick or not, because it's incapsualted in our private logic.

Reply:
If you agree to the fact that this is ugly and not welcome I would not place all this uglyness across the code base that also compromises clarity and testability.
I suggest a wrapper class for validation, this way you'll contain the "damage".

Reply:
No, it's not not welcome for libs. I was talkin only about end-projects. How does the wrapper class helps us with testability?

Reply:
You handle and test Validation exception only in the wrapper class e re-throw an internal exception (declared in THIS library) and catch the internal exception across the code. This way you need to check eventual class renames only in the wrapper and also can swap validation libraries if you want. 
The origin of the problem comes from the fact that when I installed v2 all the tests here were passing, but the library had some weird behaviours because the class in the catch sentences did not exist anymore.

Reply:
Also it looks like that there is a phpversion() function (did not know that), that could be useful to make take some decisions. Composer is php code so if they can check version, so we do. I did not check if there are any problems with thread safety, etc, with these function, but I don't see why

Reply:
see also https://www.php.net/manual/ru/reserved.constants.php
https://www.php.net/manual/ru/function.version-compare.php
very useful functions. there are also some mechanics like `class_exists`, and some utilities like https://github.com/Ocramius/PackageVersions (which was superseded by composer 2 IIUC)

Reply:
Thanks for the links, I was aware of class_exists and all other versioning functions, but never investigated on retrieving the php exe version, good to know.

Reply:
Hi,
Here's am example  for a validation wrapper, it's very rough and not tested, just to give an idea. DataValidation is a new exception class that should be created.
```
namespace League\OpenAPIValidation\Foundation;

use League\OpenAPIValidation\Foundation\Exception\DataValidation;
use Respect\Validation\Validator;

class DataValidator
{
    public function __callStatic($name, $arguments)
    {
        $v = self::getValidationMajorVersion();
        try {
            $n = $name;
            if ($name == 'numeric') {
                $v = self::getValidationMajorVersion();
                if ($v >= 2) {
                    $n = 'numericVal';
                }
            }
            call_user_func_array([Validator, $n], $arguments);
        } catch (ExceptionInterface|\Respect\Validation\Exceptions\Exception $ex) {
            throw new DataValidation($ex->getMessage(), $ex->getCode(), $ex);
        }
    }

    public static function getValidationMajorVersion()
    {
        if (version_compare(phpversion(), '7.3', '>=')) {
            return 2;
        } else {
            return 1;
        }
    }
}
```

Reply:
Yea, looks good. But actually I'd rather prefer to do check via `method_exists` since it's less error prone (imagine respect/validation for some reason fix their 1.x branch and add support of php 7.3 - the code will break).

But it still has nothing to do with testability. But yea - it's all the same trick we want to apply, just beatyfied and coupled. Good work.

Some notes though:
* `DataValidation` isn't always correct exception here, since we also utilize validation as a defensive validations for schema state (which this lib is not responsible for) and arguments (which should be validated somewhere else and thus failure here is logic exception). So I'd rather suggest name it like `InvalidValue` or so to avoid consfused of what we actually validating

Reply:
This code would fail in the most awesome why if I'm running respect/validation 1 on php 7.3 or above

Reply:
You're right, as I said it's just an example, I wrote it in 2 minutes just to express the idea

Reply:
Yea, I forgot that you can just run validation:1 on php7.3 and up too (our travis\GA prooves it). So it won't work like this

Reply:
> You're right, as I said it's just an example, I wrote it in 2 minutes just to express the idea

Sure, idea itself is good!

Reply:
If you agree with this way I can take some time to work on a new PR as soon as possible and using method_exists, etc, otherwise I'll stay on issue #102 kindly asking to support validation v2 in some way